docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -14,305 +14,194 @@ weight: 700
 
-You can also specify the SSL/TLS configuration used by the client.
+1. Check for the presence of an e-mail address via the `email` field encoded in the OAuth2 ID token.
+1. Check for the presence of an e-mail address using the [JMESPath](http://jmespath.org/examples.html) specified via the `email_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the UserInfo endpoint specified via the `api_url` configuration option.
+1. Check for the presence of an e-mail address in the `attributes` map encoded in the OAuth2 ID token. By default Grafana will perform a lookup into the attributes map using the `email:primary` key, however, this is configurable and can be adjusted by using the `email_attribute_name` configuration option.
+1. Query the `/emails` endpoint of the OAuth2 provider's API (configured with `api_url`), then check for the presence of an email address marked as a primary address.
+1. If no email address is found in steps (1-4), then the email address of the user is set to an empty string.
 
-- Set `tls_client_cert` to the path of the certificate.
-- Set `tls_client_key` to the path containing the key.
-- Set `tls_client_ca` to the path containing a trusted certificate authority list.
+### Login
 
-`tls_skip_verify_insecure` controls whether a client verifies the server's certificate chain and host name. If it is true, then SSL/TLS accepts any certificate presented by the server and any host name in that certificate. _You should only use this for testing_, because this mode leaves SSL/TLS susceptible to man-in-the-middle attacks.
+Grafana determines a user's login by using the [JMESPath](http://jmespath.org/examples.html) specified via the `login_attribute_path` configuration option.
+The order of operations is as follows:
 
-`auth_style` controls which [OAuth2 AuthStyle](https://pkg.go.dev/golang.org/x/oauth2#AuthStyle) is used when token is requested from OAuth provider. It determines how `client_id` and `client_secret` are sent to Oauth provider.
-Available values are `AutoDetect`, `InParams` and `InHeader`. By default, `AutoDetect` is used.
+1. Evaluate the `login_attribute_path` JMESPath expression against the OAuth2 ID token.
+1. Evaluate the `login_attribute_path` JMESPath expression against the JSON data obtained from UserInfo endpoint, which is specified via the `api_url` configuration option.
 
-Set `empty_scopes` to true to use an empty scope during authentication. By default, Grafana uses `user:email` as scope.
+### Display name
 
-### Email address
+You can set a user's display name with [JMESPath](http://jmespath.org/examples.html) using the `name_attribute_path` configuration option. It operates the same way as the `login_attribute_path` option.
 
-Grafana determines a user's email address by querying the OAuth provider until it finds an e-mail address:
+### Groups
```

Comment:
I'm not sure how this technical detail is important to the user. Why do they need to know this? And can we add whatever is important to them to the table instead of having a separate section?